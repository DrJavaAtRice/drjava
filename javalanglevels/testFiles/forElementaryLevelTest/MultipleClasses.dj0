//This tests that multiple classes can be defined in one file at the Elementary level.

class OneClass {
  int i;
}

class TwoClass {
  int ii;
}

class ThreeClass {
  int iii;
}