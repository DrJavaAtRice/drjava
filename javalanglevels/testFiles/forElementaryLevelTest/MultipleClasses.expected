// Language Level Converter line number map: dj*->java. Entries: 13
//     1->4         2->5         3->6         4->7         5->8         6->47        7->48        8->49   
//     9->50       10->89       11->90       12->91       13->92   
//This tests that multiple classes can be defined in one file at the Elementary level.

class OneClass {
  private final int i;

  /** This method is automatically generated by the Language Level Converter. */
  public OneClass(int i) {
    super();
    this.i = i;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int i() {
    return i;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        i() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      OneClass cast = ((OneClass) o);
      return (i() == cast.i());
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        i();
  }
}

class TwoClass {
  private final int ii;

  /** This method is automatically generated by the Language Level Converter. */
  public TwoClass(int ii) {
    super();
    this.ii = ii;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int ii() {
    return ii;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ii() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      TwoClass cast = ((TwoClass) o);
      return (ii() == cast.ii());
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        ii();
  }
}

class ThreeClass {
  private final int iii;

  /** This method is automatically generated by the Language Level Converter. */
  public ThreeClass(int iii) {
    super();
    this.iii = iii;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int iii() {
    return iii;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        iii() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      ThreeClass cast = ((ThreeClass) o);
      return (iii() == cast.iii());
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        iii();
  }
}
